platform/kube/inject/inject.go

```diff
@@ -460,6 +463,29 @@ func intoObject(c *Config, in interface{}) (interface{}, error) {
 		}
 	}
 
+	// skip customized namespaces
+	for _, ignoredNamespace := range c.IgnoredNamespaces {
+		if ignoredNamespace == "" {
+			return out, nil
+		}
+
+		if obj.GetNamespace() == ignoredNamespace {
+			// If the namesepace was defined in both IgnoredNamespaces and Namespaces,
+			// then IgnoredNamespaces will overwrite the Namespaces.
+			if c.Namespaces[0] == "" {
+				glog.V(2).Infof("The namespace %s will be ignored by sidecar initializer", ignoredNamespace)
+			} else {
+				for _, namespace := range c.Namespaces {
+					if ignoredNamespace == namespace {
+						glog.V(2).Infof("The namespace %s will be ignored by sidecar initializer", ignoredNamespace)
+						break
+					}
+				}
+			}
```

Comment:
Is it necessary to do the if/else block?  It seems you'll ignore the namespace as long as it is on the ignoredNamespace list, regardless if it is specified in c.Namespaces.

Reply:
Here the logic is that I want to make the `ignoredNamespace` has high priority than `Namespaces`.

The if/else logic is mainly used to print logs telling end user that the `ignoredNamespace` is overwriting the `Namespaces`.

We can remove the if/else logic, but then we will not able to check if the `ignoredNamespace` is overwriting the `Namespaces` or not.

Reply:
I see your point now.  If it is mainly for the purpose of logging, you may want to clarify in that log that you are overwriting the `Namespaces`.

Reply:
Sure, let's wait some comments from @ayj  ;-)